src/filters/oidc/jwks_storage.h

```diff
@@ -0,0 +1,111 @@
+#ifndef AUTHSERVICE_SRC_FILTERS_OIDC_JWKS_STORAGE_H_
+#define AUTHSERVICE_SRC_FILTERS_OIDC_JWKS_STORAGE_H_
+
+#include <spdlog/spdlog.h>
+
+#include <iostream>
+#include <memory>
+#include <mutex>
+
+#include "jwt_verify_lib/jwks.h"
+#include "src/common/http/http.h"
+
+namespace authservice {
+namespace filters {
+namespace oidc {
+
+class JwksStorage {
+ public:
+  virtual ~JwksStorage() = default;
+
+  virtual void updateJwks(const std::string& new_jwks,
+                          google::jwt_verify::Jwks::Type type) = 0;
+
+  virtual google::jwt_verify::JwksPtr jwks() = 0;
+
+ protected:
+  google::jwt_verify::JwksPtr parseJwks(const std::string& jwks,
+                                        google::jwt_verify::Jwks::Type type) {
+    auto jwks_keys = google::jwt_verify::Jwks::createFrom(jwks, type);
+    spdlog::debug("status for jwks parsing: {}, {}", __func__,
+                  google::jwt_verify::getStatusString(jwks_keys->getStatus()));
+
+    if (jwks_keys->getStatus() != google::jwt_verify::Status::Ok) {
+      spdlog::warn("{}: failed to parse new JWKs", __func__);
+    }
+
+    return jwks_keys;
+  }
+};
+
+class PermanentJwksStorageImpl : public JwksStorage {
+ public:
+  explicit PermanentJwksStorageImpl(const std::string& jwks,
+                                    google::jwt_verify::Jwks::Type type)
+      : jwks_(jwks), type_(type) {}
+
+  void updateJwks(const std::string&, google::jwt_verify::Jwks::Type) override {
+  }
+
+  virtual google::jwt_verify::JwksPtr jwks() override {
+    return parseJwks(jwks_, type_);
+  }
+
+ private:
+  std::string jwks_;
+  google::jwt_verify::Jwks::Type type_;
+};
+
+class NonPermanentJwksStorageImpl : public JwksStorage {
```

Comment:
I feel the NonPermanent is a bit confusing, it reads to me that it can be volatile. Whereas the main benefits is based on jwks uri and fetched dynamically. 

I think these two classes if named as "StaticJwksResolver" and "DynamicJwksResolver" will be easier to follow. WDTY?

Reply:
Yeah that makes sense for me.